web/src/components/vis/GroupPredictionTile.vue

```diff
@@ -0,0 +1,356 @@
+<script>
+import RocCurve from './RocCurve.vue';
+import VisTileLarge from './VisTileLarge.vue';
+import { CSVService } from '../../common/api.service';
+import plotData from './mixins/plotData';
+
+export default {
+  components: {
+    VisTileLarge,
+    RocCurve,
+  },
+
+  mixins: [plotData('roc')],
+
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      metabolites: [], // currently selected metabolites
+      group1: '',
+      group2: '',
+      metaboliteSource: 'all',
+      metaboliteSourceOptions: [
+        { value: 'all', text: 'All metabolites' },
+        { value: 'selected', text: 'Selected Metabolites' },
+      ],
+      method: 'random_forest',
+      methodOptions: [
+        { value: 'random_forest', text: 'Random Forest' },
+        { value: 'logistic_regression', text: 'Logistic Regression' },
+      ],
+      pcaData: null, // data from factor analysis endpoint
+      threshold: 0.4, // threshold for factor analysis
+      factorAnalysisFailed: false, // true if the factor analysis request fails
+    };
+  },
+  computed: {
+    // Column names (i.e. metabolites) available for selection,
+    // filtered based on current metabolite source
+    columns() {
+      if (this.metaboliteSource === 'all') {
+        return this.dataset.column.data.filter((column) => column.column_type === 'measurement')
+          .map((column) => column.column_header);
+      }
+      if (this.metaboliteSource === 'selected') {
+        return this.dataset.selectedColumns;
+      }
+      const { metabolites } = this.pcaData;
+      if (!metabolites) {
+        return [];
+      }
+      return metabolites.filter((m, i) => (this.pcaData.factor[i] === this.metaboliteSource));
+    },
+    groups() {
+      return this.dataset.groupLevels.map((level) => level.name);
+    },
+    rocData() {
+      if (!this.plot.data || !this.group1 || !this.group2 || this.metabolites.length === 0) {
+        // if any data is missing, don't draw the ROC curve
+        return { sensitivities: [], specificities: [], auc: 0 };
+      }
+      return {
+        sensitivities: this.plot.data.sensitivities,
+        specificities: this.plot.data.specificities,
+        auc: this.plot.data.auc[0],
+      };
+    },
+  },
+  watch: {
+    // get new factor analysis when threshold changes
+    threshold() {
+      this.getFactors();
+      this.metaboliteSource = 'all';
+    },
```

Comment:
I'm not really sure what the significance of `changePlotArgs` is, but the only argument I don't see it called on anywhere is the loading threshold. Does it need to be called here?

Reply:
Assuming I'm understanding how the `plotData` mixin works, calling `changePlotArgs` changes the arguments for the current analysis (ROC in this case), calls the corresponding analysis endpoint with the new arguments, and updates the `plot` object. 
The loading threshold is used for the factor analysis, which isn't an "analysis" in the same sense that the ROC analysis is, it's sort of a secondary analysis that is used by the ROC analysis/Group Prediction. It's only called once when the component is loaded initially, then every time the loading threshold changes.